gnomad/assessment/sanity_checks.py

```diff
@@ -104,78 +116,83 @@ def make_filters_sanity_check_expr(
-        f"sum_AN_{group}_{alt_groups}": hl.sum(combo_AN),
-        f"sum_nhomalt_{group}_{alt_groups}": hl.sum(combo_nhomalt),
-    }
+    annot_dict = {}
+    for subfield in ["AC", "AN", "nhomalt"]:
+        subfield_values = []
+        for x in label_combos:
+            if metric_first_label:
+                field = f"{subfield}{delimiter}{subset}{x}"
+            else:
+                field = f"{subset}{subfield}{delimiter}{x}"
+
+            if field in info_fields:
+                subfield_values.append(t.info[field])
+            else:
+                logger.info("%s is not in table's info field", field)
 
-    ht = ht.annotate(**annot_dict)
+        annot_dict[f"sum_{subfield}"] = hl.sum(subfield_values)
 
+    field_check_expr = {}
+    field_check_details = {}
     for subfield in ["AC", "AN", "nhomalt"]:
-        if not subpop:
-            generic_field_check(
-                ht,
-                (
-                    ht.info[f"{prefix}{subfield}_{group}"]
-                    != ht[f"sum_{subfield}_{group}_{alt_groups}"]
-                ),
-                f"{prefix}{subfield}_{group} = sum({subfield}_{group}_{alt_groups})",
-                [
-                    f"info.{prefix}{subfield}_{group}",
-                    f"sum_{subfield}_{group}_{alt_groups}",
-                ],
-                verbose,
+        if metric_first_label:
+            check_field_left = f"{subfield}{delimiter}{subset}{group}"
+            check_field_right = (
+                f"sum_{subfield}{delimiter}{subset}{group}{delimiter}{alt_groups}"
             )
         else:
-            generic_field_check(
-                ht,
-                (
-                    ht.info[f"{prefix}{subfield}_{subpop}_{group}"]
-                    != ht[f"sum_{subfield}_{group}_{alt_groups}"]
-                ),
-                f"{prefix}{subfield}_{subpop}_{group} = sum({subfield}_{group}_{alt_groups})",
-                [
-                    f"info.{prefix}{subfield}_{subpop}_{group}",
-                    f"sum_{subfield}_{group}_{alt_groups}",
-                ],
-                verbose,
+            check_field_left = f"{subset}{subfield}{delimiter}{group}"
+            check_field_right = (
+                f"sum_{subset}{subfield}{delimiter}{group}{delimiter}{alt_groups}"
             )
+        field_check_expr, field_check_details = make_field_check_dicts(
+            field_check_expr=field_check_expr,
+            field_check_details=field_check_details,
+            check_description=f"{check_field_left} = {check_field_right}",
+            cond_expr=t.info[check_field_left] != annot_dict[f"sum_{subfield}"],
+            display_fields=hl.struct(
+                **{
+                    check_field_left: t.info[check_field_left],
+                    f"sum_{subfield}": annot_dict[f"sum_{subfield}"],
+                }
+            ),
+        )
+
+    return field_check_expr, field_check_details
```

Comment:
PART OF LARGE SUGGESTION
```suggestion
        field_check_expr[f"{check_field_left} = {check_field_right}"] = {
            "expr": hl.agg.count_where(
                t.info[check_field_left] != annot_dict[f"sum_{subfield}"]
            ),
            "display_fields": hl.struct(
                **{
                    check_field_left: t.info[check_field_left],
                    f"sum_{subfield}": annot_dict[f"sum_{subfield}"],
                }
            ),
        }

    return field_check_expr
```